package.json

```diff
@@ -29,27 +29,28 @@
   },
   "dependencies": {
     "@instana/collector": "1.137.0",
-    "@nestjs/common": "7.6.18",
-    "@nestjs/config": "1.0.1",
-    "@nestjs/core": "7.6.18",
-    "@nestjs/platform-express": "7.6.18",
+    "@nestjs/axios": "^0.0.3",
```

Comment:
Is axios a new dependency with the v8 of nestjs?

Reply:
@MeStrak
Did you try to run the tests?
I have many typed errors with the data structure returned by Axios. I do not know why those TS errors did not pop up before this upgrade.
Most probably the best would be to address generically the `data` response structure from Confluence with a set of Interfaces.

Reply:
Hi @PiiXiieeS, that's strange. Yes I ran the tests and they were working fine, apart from the speed tests failing occasionally due to exceeding the time limit by between 10 and 60ms which I think is actually just related to the performance of my laptop.

HTTP Module is extracted to @nestjs/axios in v8. From the migration guide docs:
> HTTP module
The HttpModule exported from the @nestjs/common package has been deprecated and will be removed in the next major release. Instead, please use the @nestjs/axios package (otherwise, there are no API differences).

I just ran it again on the version you merged with the latest updates for body.view and tests are running fine + I can build and run Konviw (and the tests are running ok on the PR pipeline).

Where are you seeing the type errors you mentioned? Did you run npm install to make sure the new deps were installed correctly?

Reply:
@MeStrak 

Promise<AxiosResponse<any, any>>' is deprecated.ts(6385)
Observable.d.ts(125, 9): The declaration was marked as deprecated here.
(method) Observable<AxiosResponse<any, any>>.toPromise(): Promise<AxiosResponse<any, any>> (+2 overloads)
@deprecated — Replaced with firstValueFrom and lastValueFrom . Will be removed in v8. Details: https://rxjs.dev/deprecations/to-promise

Probably best should be to move to Observables
https://docs.nestjs.com/techniques/http-module

Reply:
Hmm, rxjs is the only dependency which isn't pinned so I wonder if that could be the reason for this. You might have a different version in your node_modules folder compared to a fresh install or npm ci.

I'll do some testing to see if I can pin that dependency too.

Reply:
@PiiXiieeS the rxjs dependency is now pinned @7.4.0.

Could you try installing dependencies with npm ci which should clear node_modules ot make sure everything is clean before the install?

Reply:
Not very clean but with this commit and casting all data responses as `any` I have fixed my type issues:
https://github.com/Sanofi-IADC/konviw/pull/288/commits/74b879d6a48ee6b49694f534aea0b82eac083b62

I have started an interface for Confluence API which is WIP and we may use in the near future to properly type the responses retrieved from the API.
https://github.com/Sanofi-IADC/konviw/pull/288/commits/7c9d263026416fe3fe7f70b710a998836d3dbba3